lumicks/pylake/tests/test_channels/test_channels.py

```diff
@@ -116,23 +116,51 @@ def test_calibration_continuous_channels():
 
 
 def test_slice_properties():
-    size = 5
-    s = channel.Slice(channel.TimeSeries(np.random.rand(size), np.random.rand(size)))
-    assert len(s) == size
-    assert s.sample_rate is None
+    size = 10
 
     s = channel.Slice(channel.Continuous(np.random.rand(size), start=0, dt=1))
     assert len(s) == size
     assert s.sample_rate == 1e9
+    timesteps = np.asarray([1], dtype=int)
+    np.testing.assert_equal(s._timesteps, timesteps)
+    np.testing.assert_equal(s.sample_rates, 1e9 / timesteps)
+
+    rng = np.random.default_rng()
+    timestamps = np.unique(np.sort(rng.integers(1e9, 10e9, size=size)))
+    s = channel.Slice(channel.TimeSeries(rng.random(len(timestamps)), timestamps))
+    assert len(s) == len(timestamps)
+    assert s.sample_rate is None
+    timesteps = np.unique(np.diff(timestamps)).astype(int)
+    np.testing.assert_equal(s._timesteps, timesteps)
+    np.testing.assert_equal(s.sample_rates, 1e9 / timesteps)
 
-    size = 10
     s = channel.Slice(channel.TimeTags(np.arange(0, size, dtype=np.int64)))
     assert len(s) == size
     assert s.sample_rate is None
+    with pytest.raises(
+        NotImplementedError,
+        match=f"`_timesteps` are not available for {s._src.__class__.__name__} data",
```

Comment:
for the tests, I would hard code in the string here and below rather than using the f-string; I think it's cleaner to read

Reply:
Done